dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/viewtools/content/ContentToolTest.java

```diff
@@ -414,4 +423,176 @@ private ContentTool getContentTool(final long languageId){
+        new ContentletDataGen(contentType.id())
+                .setPolicy(IndexPolicy.FORCE)
+                .setProperty(SYS_PUBLISH_DATE, contentPublishDate.getTime())
+                .nextPersisted();
+
+        final String query = String.format(QUERY_BY_STRUCTURE_NAME, contentType.variable());
+
+        final ContentTool contentTool = getContentTool(null);
+
+        final PaginatedContentList<ContentMap> contents = contentTool.pullPerPage(query, 1, 2, null);
+        assertEquals(0, contents.size());
+    }
+
+    /**
+     * Method to test: {@link ContentTool#pullPerPage(String, int, int, String)}
+     * When: there is a content with a publish date set to tomorrow and the time machine date is the date after tomorrow
+     * Should: return one content
+     */
+    @Test
+    public void whenTheTimeMachineDateAndPublishDateAreTomorrowShouldReturnOneContent() {
+        final Calendar publishDate = Calendar.getInstance();
+        publishDate.add(Calendar.DATE, 1);
+
+        final Calendar timeMachine = Calendar.getInstance();
+        timeMachine.add(Calendar.DATE, 2);
+
+        final ContentType contentType = TestDataUtils.getNewsLikeContentType();
+        final Contentlet contentlet = new ContentletDataGen(contentType.id())
+                .setProperty(SYS_PUBLISH_DATE, publishDate.getTime())
+                .languageId(1)
+                .nextPersisted();
+
+        final String query = String.format(QUERY_BY_STRUCTURE_NAME, contentType.variable());
+
+        final ContentTool contentTool = getContentTool(timeMachine);
+
+        final PaginatedContentList<ContentMap> contents = contentTool.pullPerPage(query, 1, 2, null);
+
+        assertEquals(1  , contents.size());
+        assertEquals(1  , contents.getTotalResults());
+        assertEquals(contentlet.getIdentifier(), contents.get(0).getContentObject().getIdentifier());
+    }
+
+    @NotNull
+    private ContentTool getContentTool(Calendar timeMachine) {
+        final ContentTool contentTool  = new ContentTool();
+
+        final String time = timeMachine != null ? Long.toString(timeMachine.getTime().getTime()) : null;
+
+        final HttpSession session = mock(HttpSession.class);
+        final HttpServletRequest request = mock(HttpServletRequest.class);
+        when(request.getSession(false)).thenReturn(session);
+        when(request.getSession()).thenReturn(session);
+        when(request.getAttribute(WebKeys.USER)).thenReturn(APILocator.systemUser());
+
+        when(session.getAttribute("tm_date")).thenReturn(time);
+
+        final ViewContext viewContext = mock(ViewContext.class);
+        when(viewContext.getRequest()).thenReturn(request);
+
+        contentTool.init(viewContext);
+        return contentTool;
+    }
+
+    /**
+     * Method to test: {@link ContentTool#pullPerPage(String, int, int, String)}
+     * When: there is a content with a expire  date set to tomorrow and the time machine date is the date after tomorrow
```

Comment:
code expresses a different thing that this javadoc

Reply:
done https://github.com/dotCMS/core/pull/18863/commits/a73fc1f7ed898352425c4be3a6211ffafdea42a6#diff-b77a02567ac2b8e298742516ca666f0dR510